### DISCUSSION TITLE
TurbSim: turbolent wind for wind farm control

### DISCUSSION THREAD

**Discussion by Alessandro5I** [OpenFAST/openfast#2440] [General]
Hi all,
I'm writing an algorithm for wind farm control.
I'm wondering if it is possible to create a turbolent wind file for FAST.Farm with a varying mean wind speed and wind direction in the order of 20 - 30 minutes.
I would like to have the mean wind speed and direction as in the attached picture. 
![Screenshot 2024-09-26 203739](https://github.com/user-attachments/assets/3a2a4b60-44aa-4b27-9f86-8fdf1881a21a)
![Screenshot 2024-09-26 203842](https://github.com/user-attachments/assets/0f3458b5-a116-49b0-8599-d24cee3b0c35)
Is this possible? If yes, what is the best way to reach it?
Thanks,
Alessandro

**Comment by jjonkman** [comment]
Dear @Alessandro5I,

Can you clarify your example?  You mention that you want a 20-30 minute simulation, but your example shows wind speed and direction over 70 hours.  Can you clarify what the source of your wind data you want to use is?

FAST.Farm allows turbulent inflow data to be specified both using synthetic turbulence and high-fidelity precursor simulation.  The synthetic turbulence generator that NREL has released (TurbSim) generates turbulent wind about a fixed mean wind speed and direction.  But if you are using a high-fidelity precursor to generate inflow for FAST.Farm (such as AMR-Wind or SOWFA), this precursor could include time-varying wind speed and direction changes.

Best regards,
